pybamm/solvers/processed_variable.py

```diff
@@ -54,7 +57,27 @@ def __init__(self, base_variable, solution, known_evals=None):
         self.domain = base_variable.domain
         self.auxiliary_domains = base_variable.auxiliary_domains
         self.known_evals = known_evals
-
+        self.warn = warn
+
+        # Set timescale
+        try:
+            self.timescale = solution.model.timescale_eval
+        except AttributeError:
+            if self.warn:
+                pybamm.logger.warning("No timescale provided. Using default of 1 [s]")
+            self.timescale = 1
```

Comment:
A model should always have a timescale, since the base model has the default timescale of 1s

Reply:
there is a bit of an issue with the shape when the timescale depends on the inputs. In `base_solver` the timescale is evaluated with the initial inputs. if you then try to do `model.timescale.evaluate(inputs=solution.inputs)` in `ProcessedVariable` then `inputs` is now of length `t`. For now I've done a try/except with using `timescale_eval` and then evaluating if that isn't found. Another option would be to store the initial values of the inputs in the solution and use that (since the timescale *should* be the same for all times).

Reply:
I think the safest option is to not allow the timescale to depend on inputs. i.e. do `model.timescale.evaluate()` instead of `model.timescale.evaluate(inputs=solution.inputs)` (and use a try except for a clear error message). Allowing the timescale to change in e.g. parameter fitting runs is probably asking for trouble

Reply:
yeah I agree. I've added a check for this in `base_solver`